js/gemini.js

```diff
@@ -278,12 +278,14 @@ module.exports = class gemini extends Exchange {
 
     async fetchOpenOrders (symbol = undefined, since = undefined, limit = undefined, params = {}) {
         await this.loadMarkets ();
-        let market = undefined;
+        let response = await this.privatePostOrders (params);
+        let orders = this.parseOrders (response, undefined, since, limit);
         if (typeof symbol !== 'undefined') {
-            market = this.market (symbol);
+            if (!(symbol in this.markets))
+                throw new ExchangeError (this.id + ' has no symbol ' + symbol);
```

Comment:
This particular check + exception isn't needed, as it will most likely break earlier. I'd just pass it into filterBySymbol here.

Actually, I wonder why would you call this.filterBySymbol manually here, as it should already be handled in this.parseOrders (it has a call to filter them inside). So the old code should have worked, but you said there is a bug here that I can't see. Will be thankful for your comment on this.

Reply:
I can skip the check for sure.  As for the bug, it's a slightly tricky one -- you get all of the open orders (regardless of symbol) from `privatePostOrders`, so when you pass a market into `parseOrders` it will pass that market into `parseOrder`, which assumes that you've provided the correct market and uses the symbol from that market instead of searching for the right one (see https://github.com/ccxt/ccxt/blob/master/js/gemini.js#L248 for more info).  So in the end you'll ask for the orders on 'BTC/USD' but you get every order with their normal currency overridden to 'BTC/USD'.  I noticed this bug while doing the work on Huobi.  Does that make sense?